### PR TITLE
[Fix] Don not crash when research is corrupted  

### LINKED ISSUE #9005: Crash when viewing broken research in research table
When placing a specific copy of a "Wand Focus: Crafting" research into the research table, a crash occurs. It looks like this research was previously corrupted somehow unless I'm misunderstanding a game mechanic, as it has a type 1 tile with no listed aspect. It appears that it cannot be completed in regular Thaumcraft, but the game doesn't crash either.

Faulty NBT data in the hexgrid:
faulty-nbt

Here is the research when opened in the regular research table:
corrupted-research

Here is what a newly created research looks like for "Wand Focus: Crafting":
normal-research

[fml-client-latest.log](https://github.com/GTNewHorizons/thaumcraft-research-tweaks/files/7656355/fml-client-latest.log)

Open in NBT explorer to view the data of the research which triggers the crash (zipped so I could upload it) :
[faulty-research.zip](https://github.com/GTNewHorizons/thaumcraft-research-tweaks/files/7656360/faulty-research.zip)

### PR DESCRIPTION
**Pull request by TheElan**
Fixes https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/9005

Also switched to GTNH forge gradle (this required adding some repos).

In rare case, when save gets corrupted and occupied hex is missing aspect in tags should show: 
image

The only fix at that point is manually fixing save file or making new research notes.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -48,20 +52,24 @@ repositories {
...
-        name = "minecraftforge"
-        url = "https://files.minecraftforge.net/maven/"
+        name = "Overmind forge repo mirror"
+        url = "https://gregtech.overminddl1.com/"
     }
     maven {
-        name = "jeknins-thirdparty"
-        url = "http://jenkins.usrv.eu:8081/nexus/content/repositories/thirdparty/"
+        name = "forge"
+        url = "https://files.minecraftforge.net/maven"
```
Comment (bombcar) [inline build.gradle]: Isn't this https://maven.minecraftforge.net no?

Comment (TheElan) [reply build.gradle]: Looks like this url is working again, but I'll change anyway.

Comment (TheElan) [reply build.gradle]: done

**Review by Glease — APPROVED** [review]
nothing suspicious. looks ok

**Pull request merged**